main/modes/games/ultimateTTT/ultimateTTTgame.c

```diff
@@ -14,143 +26,182 @@ typedef void (*cursorFunc_t)(ultimateTTT_t* ttt);
 // Function Declarations
 //==============================================================================
 
-static void tttPlacePiece(ultimateTTT_t* ttt, const vec_t* subgame, const vec_t* cell, tttPlayer_t piece);
+static void tttPlaceMarker(ultimateTTT_t* ttt, const vec_t* subgame, const vec_t* cell, tttPlayer_t marker);
 static tttPlayer_t checkWinner(ultimateTTT_t* ttt);
 static tttPlayer_t checkSubgameWinner(tttSubgame_t* subgame);
-static wsg_t* getPieceWsg(ultimateTTT_t* ttt, tttPlayer_t p, bool isBig);
+static wsg_t* getMarkerWsg(ultimateTTT_t* ttt, tttPlayer_t p, bool isBig);
+static playOrder_t tttGetPlayOrder(ultimateTTT_t* ttt);
 
 //==============================================================================
 // Functions
 //==============================================================================
 
 /**
- * @brief TODO
+ * @brief Start a multiplayer game. This is called after a connection is established
  *
- * @param ttt
+ * @param ttt The entire game state
  */
 void tttBeginGame(ultimateTTT_t* ttt)
 {
-    ttt->ui = TUI_GAME;
+    // Set the state as not playing yet
+    ttt->game.state = TGS_NOT_PLAYING;
 
-    // If going first
-    if (GOING_FIRST == p2pGetPlayOrder(&ttt->p2p))
+    // Reset the board
+    for (int16_t y = 0; y < 3; y++)
     {
-        // Set own piece type
-        ttt->p1Piece = TTT_PIECE_X;
+        for (int16_t x = 0; x < 3; x++)
+        {
+            ttt->game.subgames[x][y].winner = TTT_NONE;
+            for (int16_t sy = 0; sy < 3; sy++)
+            {
+                for (int16_t sx = 0; sx < 3; sx++)
+                {
+                    ttt->game.subgames[x][y].game[sx][sy] = TTT_NONE;
+                }
+            }
+        }
+    }
 
-        // Send piece type to other swadge
-        tttMsgSelectPiece_t sel = {
-            .type  = MSG_SELECT_PIECE,
-            .piece = ttt->p1Piece,
-        };
-        p2pSendMsg(&ttt->p2p, (const uint8_t*)&sel, sizeof(sel), tttMsgTxCbFn);
+    // Reset the cursor
+    ttt->game.cursor.x          = 0;
+    ttt->game.cursor.y          = 0;
+    ttt->game.selectedSubgame.x = 0;
+    ttt->game.selectedSubgame.y = 0;
+    ttt->game.cursorMode        = SELECT_SUBGAME;
+
+    // Default indices
+    ttt->game.p1MarkerIdx = 0;
+    ttt->game.p2MarkerIdx = 0;
+
+    // Clean up after showing instructions
+    ttt->game.p2p.cnc.playOrder = NOT_SET;
```

Comment:
I think this (or a related change) is breaking multiplayer. When I try to start the game, both sides connect immediately and then both sides show a blue square around the top-left game, which neither swadge is able to move. I did a git bisect and it looks like it started after 59739eb580b6b6a06857343e103f4305b308b7b6. I looked around a bit and don't see anywhere else outside of `ultimateTTThowTo.c` that actually sets it.

Reply:
And by "it" I mean `playOrder`

Reply:
It sure is.